Pass threshold to collected metrics in scheduler perf.

Argh4k opened the pull request:
/sig scheduling

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

With this change, the metrics for which the threshold is defined, will have a "Threshold" data point that should be visible in perf dash.

#### Which issue(s) this PR is related to:

N/A

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

macsko commented on test/integration/scheduler_perf/scheduler_perf.go:
```diff
@@ -1542,6 +1553,7 @@ func stopCollectingMetrics(tCtx ktesting.TContext, collectorCtx ktesting.TContex
 		if err != nil {
 			tCtx.Errorf("op %d: %s", opIndex, err)
 		}
+		addThresholdToData(items, threshold, tms)
```
Nvm, you're right

macsko commented:
/cc @sanposhiho @pohly

sanposhiho commented:
I'd leave /approve to @macsko because he already had several reviews, but I'm ok with this approach.

Argh4k committed:
Pass threshold to collected metrics in scheduler perf.

With this change, the metrics for which the threshold is defined, will have a "Threshold" data point that should be visible in perf dash.

macsko commented:
@Argh4k Why this got closed?

Argh4k commented:
Sorry,  I missed that I created this pr of my master branch instead of some feature branch and I reset it when working on other things.  Reopening.

Argh4k commented:
/test pull-kubernetes-unit

volume manager flake.

Argh4k commented:
/retest-required

Argh4k commented on test/integration/scheduler_perf/scheduler_perf_test.go:
```diff
@@ -689,3 +690,220 @@ func TestCompareMetricWithThreshold(t *testing.T) {
+			expectedThresholdName:                 "AverageThreshold",
+		},
+		{
+			name:           "value is above threshold",
```
Fixed

Argh4k commented on test/integration/scheduler_perf/scheduler_perf_test.go:
```diff
@@ -689,3 +690,220 @@ func TestCompareMetricWithThreshold(t *testing.T) {
+		expectedThresholdName                 string
+	}{
+		{
+			name:           "value is below threshold",
```
Fixed

brejman commented on test/integration/scheduler_perf/scheduler_perf.go:
```diff
@@ -1481,6 +1481,8 @@ func compareMetricWithThreshold(items []DataItem, threshold float64, metricSelec
 			continue
 		}
+		thresholdItemName := dataBucket + "Threshold"
+		item.Data[thresholdItemName] = threshold
```
Seems a bit unexpected to me that we're modifying the `item.Data` in a `compare` method

tosi3k replied:
Right now, if we have 2+ data items that do not satisfy the threshold, only the first one in the slice would be extended with the threshold data point. Is that intended?

Argh4k commented on test/integration/scheduler_perf/scheduler_perf_test.go:
```diff
@@ -672,18 +674,264 @@ func TestCompareMetricWithThreshold(t *testing.T) {
+			},
+			wantErr:  true,
+			errCount: 2,
+		},
```
Added "threshold value is added for all matching entries even with error" test case.

Argh4k replied:
As for this test case it's specifically testing applyThreshold, so I would leave as is.
